File: features/lead/src/main/java/com/nivasafinance/features/leadtasks/entity/LeadTask.java
package com.nivasafinance.features.leadtasks.entity;

import com.nivasafinance.common.audit.IdentifiableEntity;
import io.hypersistence.utils.hibernate.type.json.JsonType;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.JdbcTypeCode;
import org.hibernate.annotations.Type;
import org.hibernate.type.SqlTypes;

@Entity
@Table(name = "n_lead_tasks")
@NoArgsConstructor
@AllArgsConstructor
@Data
@Builder
@EqualsAndHashCode(callSuper = true)
public class LeadTask extends IdentifiableEntity {

    @Column(name = "lead_id", nullable = false)
    private Long leadId;

    @Column(name = "task_id", nullable = false)
    private Long taskId;

    @Type(JsonType.class)
    @JdbcTypeCode(SqlTypes.JSON)
    @Column(name = "task_details", columnDefinition = "jsonb")
    private TaskDetails taskDetails;

    @Data
    @NoArgsConstructor
    @AllArgsConstructor
    @Builder
    public static class TaskDetails {
        private String stageKey;
    }
}

File: features/tasks/src/main/java/com/nivasafinance/features/tasks/entity/LeadTask.kt
package com.nivasafinance.features.tasks.entity

import audit.AuditableEntity
import com.nivasafinance.features.tasks.enum.TaskStatus
import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.EnumType
import jakarta.persistence.Enumerated
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.Table
import org.hibernate.annotations.JdbcTypeCode
import org.hibernate.type.SqlTypes
import java.time.LocalDateTime
import java.util.UUID

@Entity
@Table(name = "lead_tasks")
data class LeadTask(
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    val id: UUID? = null,

    @Column(name = "task_key", nullable = false)
    val taskKey: String,

    @Column(name = "lead_id", nullable = false)
    val leadId: UUID,

    @Column(name = "stage", nullable = false)
    val stage: String,

    @JdbcTypeCode(SqlTypes.JSON)
    @Column(name = "task_data", columnDefinition = "jsonb")
    var taskData: Map<String, Any>? = null,

    @Column(name = "assigned_to")
    var assignedTo: String? = null,

    @Enumerated(EnumType.STRING)
    @Column(name = "status", nullable = false)
    var status: TaskStatus,

    @Column(name = "due_at")
    var dueAt: LocalDateTime? = null,

    @Column(name = "completed_at")
    var completedAt: LocalDateTime? = null,

    @Column(name = "rescheduled_at")
    var rescheduledAt: LocalDateTime? = null
) : AuditableEntity()
